### PR TITLE
AutoBumper Addons List

### PR DESCRIPTION
**Pull request by emma144**
Hello! I've come across this extension recently and I thought it would've been cool if this was added. It is an extension that adds a button at the bottom of the page on all Scratch topics that, once clicked, will send a bump message. The bump message is, of-course, customizable in the settings. Please let me know if I have missed anything! I also tested this locally and it works.

### CODE REVIEW

File: addons/auto-bumper/addon.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "Auto-Bumper",
```
Comment (DNin01) [inline addons/auto-bumper/addon.json:2]: With this name, it sounds like topics could be bumped without user interaction. A more fitting name might be:
```suggestion
  "name": "One-click bump",
```

**Comment by BroJac5246** [comment]
Suggestion: the button should only show if there hasn't been a post in a certain amount of time. This will prevent bumping when it shouldn't be done. The period could be a setting, or it could be hard-coded to a set period, such as 24 hours.

**Comment by emma144** [comment]
> Suggestion: the button should only show if there hasn't been a post in a certain amount of time. This will prevent bumping when it shouldn't be done. The period could be a setting, or it could be hard-coded to a set period, such as 24 hours.

I am not sure how I could do that, and I believe it would also be a bit complicated

**Comment by WorldLanguages** [comment]
@emma144 Congrats on making your own addon!

**Comment by WorldLanguages** [comment]
Personally I'm doubtful on whether this belongs on Scratch Addons. We could be indirectly encouraging spam and accidental bump posts.

**Comment by emma144** [comment]
Hmm, well, we could find a way to add a cooldown and a confirmation box?

**Comment by BroJac5246** [comment]
> Hmm, well, we could find a way to add a cooldown and a confirmation box?

The cooldown should be the 60/120- second rule; the confirmation box would probably be advisable to help prevent missclicks.

**Comment by emma144** [comment]
> > Hmm, well, we could find a way to add a cooldown and a confirmation box?
> 
> The cooldown should be the 60/120- second rule; the confirmation box would probably be advisable to help prevent missclicks.

The cooldown is currently the 120/60 second post rule already

**Comment by BroJac5246** [comment]
> > > Hmm, well, we could find a way to add a cooldown and a confirmation box?
> > 
> > The cooldown should be the 60/120- second rule; the confirmation box would probably be advisable to help prevent missclicks.
> 
> The cooldown is currently the 120/60 second post rule already

I'm personally cool with it staying at that

**Comment by WorldLanguages** [comment]
> Personally I'm doubtful on whether this belongs on Scratch Addons. We could be indirectly encouraging spam and accidental bump posts.

Note that by "spam" here, I was not only referring to bumping often. Simply bumping 1 forum post unnecessarily would also count as spam.
If we release this addon to the public we will be indirectly encouraging users to bump their posts because it will be a bit easier. It's also possible that some users might learn about the concept of bumping because of our addon explanation.

**Comment by WorldLanguages** [comment]
I'm not a hardcore forums user, but I believe there are some forum categories where bumping makes no sense and is disallowed, right? Would we need to consider those cases?
I don't like the fact publishing this addon could possibly mean we need to educate its users about forum etiquette regarding bumping.

**Comment by BroJac5246** [comment]
> > Personally I'm doubtful on whether this belongs on Scratch Addons. We could be indirectly encouraging spam and accidental bump posts.
> 
> Note that by "spam" here, I was not only referring to bumping often. Simply bumping 1 forum post unnecessarily would also count as spam.

[This topic](https://scratch.mit.edu/discuss/topic/395720/) suggests that you shouldn't bump in the following situations:
```
1. Bumping more than twice a day. This is considered spam.
2. Bumping really old topics which are no longer useful. (Like old questions in which the owner is no longer looking for answer)
3. Stickies (These are already at the top so don't need to be bumped). 
```

The first can be fixed by only showing the button on topics that haven't received a post in 24 hours; the third can be fixed by not showing the button on stickies. The second would be hard to determine, but we could also show a warning if the topic hasn't received a post in, say, 6 months.

If we also include a note in the settings menu then this should be okay.

**Comment by WorldLanguages** [comment]
I don't like where this is going. We need to implement some of these rules into the code, and educate the user about others

**Comment by emma144** [comment]
Actually, @WorldLanguages, it seems complicated and from a second thought is not very practical

**Comment by WorldLanguages** [comment]
If anyone wants to re-discuss the addon idea, please create a "new addon" issue.

**Comment by GrahamSH-LLK** [comment]
> > > Personally I'm doubtful on whether this belongs on Scratch Addons. We could be indirectly encouraging spam and accidental bump posts.
> > 
> > 
> > Note that by "spam" here, I was not only referring to bumping often. Simply bumping 1 forum post unnecessarily would also count as spam.
> 
> [This topic](https://scratch.mit.edu/discuss/topic/395720/) suggests that you shouldn't bump in the following situations:
> 
> ```
> 1. Bumping more than twice a day. This is considered spam.
> 2. Bumping really old topics which are no longer useful. (Like old questions in which the owner is no longer looking for answer)
> 3. Stickies (These are already at the top so don't need to be bumped). 
> ```
> 
> The first can be fixed by only showing the button on topics that haven't received a post in 24 hours; the third can be fixed by not showing the button on stickies. The second would be hard to determine, but we could also show a warning if the topic hasn't received a post in, say, 6 months.
> 
> If we also include a note in the settings menu then this should be okay.

Just a comment, @BroJac5246, you aren't a maintainer of this project by any sense of the term. You keep giving people a sense that _you_ approve and _you're_ the person who has to approve. The problem is that that's not true. Just a thought.

**Comment by BroJac5246** [comment]
> Just a comment, @BroJac5246, you aren't a maintainer of this project by any sense of the term. You keep giving people a sense that _you_ approve and _you're_ the person who has to approve. The problem is that that's not true. Just a thought.

Oops, thank you for pointing that out.

**Pull request closed without merging**